arm-machinelearning/2016-05-01-preview/swagger/commitmentPlans.json

```diff
@@ -0,0 +1,903 @@
+                        "in": "path",
+                        "required": true,
+                        "type": "string"
+                    },
+                    {
+                        "name": "$skipToken",
+                        "description": "Continuation token for pagination.",
+                        "in": "query",
+                        "required": false,
+                        "type": "string"
+                    },
+                    {
+                        "$ref": "#/parameters/APIVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/PaginatedResult[PlanUsageHistory]"
+                        }
+                    }
+                }
+            }
+        }
+    },
+    "parameters": {
+        "SubscriptionIdParameter": {
+            "name": "subscriptionId",
+            "description": "Azure Subscription ID.",
+            "in": "path",
+            "type": "string",
+            "required": true
+        },
+        "APIVersionParameter": {
+            "in": "query",
+            "name": "api-version",
+            "type": "string",
+            "description": "The versiong of the Microsoft.MachineLearning resource provider API to be used.",
+            "required": true
+        }
+   },
+   "definitions": {
+        "Resource": {
+            "x-ms-azure-resource": true,
+            "required": [
+                "location"
+            ],
+            "properties": {
+                "id": {
+                    "description": "Resource Id",
+                    "type": "string",
+                    "readOnly": true
+                },
+                "name": {
+                    "description": "Resource Name",
+                    "type": "string"
+                },
+                "location": {
+                    "description": "Resource Location",
+                    "type": "string"
+                },
+                "type": {
+                    "description": "Resource type",
+                    "type": "string",
+                    "readOnly": true
+                },
+                "tags": {
+                    "description": "Resource tags",
+                    "type": "object",
+                    "additionalProperties": {
+                        "type": "string"
+                    }
+                }
+            }
+        },
+        "CommitmentAssociation": {
+            "type": "object",
+            "description": "Represents the association between a commitment plan and some other resource, such as a Machine Learning web service.",
+            "properties": {
+                "etag": {
+                    "type": "string",
+                    "description": "An entity tag used to enforce optimistic concurrency."
+                },
+                "id": {
```

Comment:
CommitmentAssociation should inherit "allOf" from the Resource. In that way the Resource properties need not be defined again in this class.